### PR TITLE
Update minimum eslint version

### PR DESCRIPTION
**Pull request by SimenB**
Same as #6, but includes update to eslint@3 and eslint-plugin-react@6 (still in alpha).

This has conflicts against #12, but I'll deal with them after the merge of either 😄 

Breaking in eslint@3 is mostly just dropped support for node@<4. Breaking in react plugin 5 is what deprecation warnings there are (you have to set the version of react you're using in config). Breaking in 6 is just to play safe with dual eslint@2 and eslint@3 support, as well as explicit peer dep on eslint.

https://github.com/yannickcr/eslint-plugin-react/blob/master/CHANGELOG.md

### CODE REVIEW

File: env-es6.js [hunk 1]
```diff
@@ -6,14 +6,14 @@ module.exports = {
...
-        'no-this-before-super': 2,
-        'no-var': 2,
-        // TODO: bring back to 2 when object-shorthand rule supports ES7 object spread.
+        'constructor-super': 'error',
+        'generator-star-spacing': 'off',
+        'generator-star': 'off',
+        'no-this-before-super': 'error',
+        'no-var': 'error',
+        // TODO: bring back to 'error' when object-shorthand rule supports ES7 object spread.
         // cf. https://github.com/eslint/eslint/issues/2486
```
Comment (SimenB) [inline env-es6.js]: this was fixed over a year ago, btw

File: env-es6.js [hunk 2]
```diff
@@ -6,14 +6,14 @@ module.exports = {
...
-        'object-shorthand': 1,
-        'prefer-const': 0
+        'object-shorthand': 'warn',
```
Comment (nosilleg) [inline env-es6.js]: Feel free to update this to error and remove the comment.

File: package.json [hunk 3]
```diff
@@ -8,12 +8,12 @@
     "test": "eslint ."
   },
   "devDependencies": {
-    "eslint": "^2.0.0",
-    "eslint-plugin-react": "^4.0.0"
+    "eslint": "^3.1.1",
+    "eslint-plugin-react": "^6.0.0-alpha.1"
```
Comment (nosilleg) [inline package.json:12]: Do you expect this to hit final before we do a release of v3?  Otherwise I would go with v5.

File: env-es6.js (hunk 2 above)
Comment (SimenB) [inline env-es6.js]: will do

File: package.json (hunk 3 above)
Comment (SimenB) [inline package.json:12]: v5 doesn't support eslint@3 unfortunately

Comment (SimenB) [inline package.json:12]: But yes, I expect it to be stable before we do v3, especially if you want to also make som rule changes and not just dep updates 😄

**Pull request merged**